dvc/stage/__init__.py

```diff
@@ -172,11 +174,11 @@ def __eq__(self, other):
             and self.path_in_repo == other.path_in_repo
         )
 
-    @property
+    @cached_property
     def path_in_repo(self):
         return relpath(self.path, self.repo.root_dir)
 
-    @property
+    @cached_property
```

Comment:
There's a test `test_stage_strings_representation` that chdir to a different directory and checks for `relpath`. 

https://github.com/iterative/dvc/blob/a0c229cf9b9a33c1301c0ce0727a4d89fb73aaa1/tests/func/test_stage.py#L222